### PR TITLE
Bump csi-provisioner to v2.1.1

### PR DESCRIPTION
**Pull request by croomes**
Bumps the default version of the CSI provisioner, which we fork: https://github.com/storageos/external-provisioner/tree/v2.1.1-patched

v2.1.1 requires k8s 1.17+.

### CODE REVIEW

**Review by mhmxs — APPROVED** [review]

**Review by darkowlzz — APPROVED** [review]

File: internal/pkg/image/image.go [hunk 1]
```diff
@@ -9,7 +9,7 @@ const (
 	CSIv1ClusterDriverRegistrarContainerImage = "quay.io/k8scsi/csi-cluster-driver-registrar:v1.0.1"
 	CSIv1NodeDriverRegistrarContainerImage    = "quay.io/k8scsi/csi-node-driver-registrar:v1.2.0"
 	CSIv1ExternalProvisionerContainerImageV1  = "storageos/csi-provisioner:v1.4.0"
```
Comment (darkowlzz) [inline internal/pkg/image/image.go:11]: We can remove this constant completely. Not being used anywhere.

**Comment by darkowlzz** [comment]
v2.1.2 was release 2 days ago with a minor logging change https://github.com/kubernetes-csi/external-provisioner/compare/v2.1.1...v2.1.2 .

File: internal/pkg/image/image.go (hunk 1 above)
Comment (croomes) [reply internal/pkg/image/image.go]: done

**Comment by croomes** [comment]
> v2.1.2 was release 2 days ago with a minor logging change [kubernetes-csi/external-provisioner@v2.1.1...v2.1.2](https://github.com/kubernetes-csi/external-provisioner/compare/v2.1.1...v2.1.2) .

I think it's ok to leave this for the next update.

**Pull request merged**